### PR TITLE
feat: limit-req support nodelay

### PR DESCRIPTION
**Pull request by nanamikon**
### What this PR does / why we need it:
limit-req support nodelay, refer to (#4382)

### Pre-submission checklist:

* [1] Did you explain what problem does this PR solve? Or what new features have been added?
* [1] Have you added corresponding test cases?
* [1] Have you modified the corresponding document?
* [1 ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: t/plugin/limit-req2.t [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+
+=== TEST 4: the second request will not timeout because of nodelay
+--- abort
+--- timeout: 500ms
+--- pipelined_requests eval
+["GET /hello", "GET /hello"]
+--- error_code eval
+[200, 200]
```
Comment (tokers) [inline t/plugin/limit-req2.t:121]: Please check the editor, the EOL symbol is required.

Comment (nanamikon) [reply t/plugin/limit-req2.t]: fix it,   I find that reindex can not add  EOL symbol

File: docs/en/latest/plugins/limit-req.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ limit request rate using the "leaky bucket" method.
 | burst         | integer | required    |         | burst >= 0                                                               | the number of excessive requests per second allowed to be delayed. Requests exceeding this hard limit will get rejected immediately.                                      |
 | key           | string  | required    |         | ["remote_addr", "server_addr", "http_x_real_ip", "http_x_forwarded_for", "consumer_name"] | the user specified key to limit the rate, now accept those as key: "remote_addr"(client's IP), "server_addr"(server's IP), "X-Forwarded-For/X-Real-IP" in request header, "consumer_name"(consumer's username). |
 | rejected_code | integer | optional    | 503     | [200,...,599]                                                            | The HTTP status code returned when the request exceeds the threshold is rejected.                                                                      |
+| nodelay       | boolean | optional    | false   |                                                                          | If nodelay flag is true, the specified request will not get delayed  |
```
Comment (tokers) [inline docs/en/latest/plugins/limit-req.md:45]:
```suggestion
| nodelay       | boolean | optional    | false   |                                                                          | If nodelay flag is true, bursted requests will not get delayed  |
```

File: docs/zh/latest/plugins/limit-req.md [hunk 3]
```diff
@@ -42,6 +42,7 @@ title: limit-req
 | burst         | integer | 必须   |        | burst >= 0                                                              | t请求速率超过 （`rate` + `brust`）的请求会被直接拒绝。                                                                                            |
 | key           | string  | 必须   |        | ["remote_addr", "server_addr", "http_x_real_ip", "http_x_forwarded_for", "consumer_name"] | 用来做请求计数的依据，当前接受的 key 有："remote_addr"(客户端IP地址), "server_addr"(服务端 IP 地址), 请求头中的"X-Forwarded-For" 或 "X-Real-IP"，"consumer_name"(consumer 的 username)。 |
 | rejected_code | integer | 可选   | 503    | [200,...,599]                                                              | 当请求超过阈值被拒绝时，返回的 HTTP 状态码。                                                                                                        |
+| nodelay       | boolean | 可选   | false  |                                                                         | 如果nodelay为false， 则满足条件的请求不会加上延迟 |
```
Comment (tokers) [inline docs/zh/latest/plugins/limit-req.md:45]: `满足条件` is too general, could you use some more detailed descriptions?

File: docs/zh/latest/plugins/limit-req.md [hunk 4]
```diff
@@ -42,6 +42,7 @@ title: limit-req
 | burst         | integer | 必须   |        | burst >= 0                                                              | t请求速率超过 （`rate` + `brust`）的请求会被直接拒绝。                                                                                            |
 | key           | string  | 必须   |        | ["remote_addr", "server_addr", "http_x_real_ip", "http_x_forwarded_for", "consumer_name"] | 用来做请求计数的依据，当前接受的 key 有："remote_addr"(客户端IP地址), "server_addr"(服务端 IP 地址), 请求头中的"X-Forwarded-For" 或 "X-Real-IP"，"consumer_name"(consumer 的 username)。 |
 | rejected_code | integer | 可选   | 503    | [200,...,599]                                                              | 当请求超过阈值被拒绝时，返回的 HTTP 状态码。                                                                                                        |
+| nodelay       | boolean | 可选   | false  |                                                                         | 如果nodelay为true， 请求速率超过 `rate` 但没有超过 （`rate` + `brust`）的请求不会加上延迟, 如果是false，则会加上延迟 |
```
Comment (Firstsawyou) [inline docs/zh/latest/plugins/limit-req.md:45]:
```suggestion
| nodelay       | boolean | 可选   | false  |                                                                         | 如果 nodelay 为 true， 请求速率超过 `rate` 但没有超过 （`rate` + `brust`）的请求不会加上延迟, 如果是 false，则会加上延迟。 |
```

**Review by tokers — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**